src/sentry/data_export/utils.py

```diff
@@ -10,46 +10,64 @@
-    except snuba.QueryOutsideRetentionError as error:
-        metrics.incr("dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0)
-        logger.info("dataexport.error: {}".format(six.text_type(error)))
-        capture_exception(error)
-        raise ExportError("Invalid date range. Please try a more recent date range.")
-    except snuba.QueryIllegalTypeOfArgument as error:
-        metrics.incr("dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0)
-        logger.info("dataexport.error: {}".format(six.text_type(error)))
-        capture_exception(error)
-        raise ExportError("Invalid query. Argument to function is wrong type.")
-    except snuba.SnubaError as error:
-        metrics.incr("dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0)
-        logger.info("dataexport.error: {}".format(six.text_type(error)))
-        capture_exception(error)
-        message = "Internal error. Please try again."
-        if isinstance(
-            error,
-            (
-                snuba.RateLimitExceeded,
-                snuba.QueryMemoryLimitExceeded,
-                snuba.QueryTooManySimultaneous,
-            ),
-        ):
-            message = "Query timeout. Please try again. If the problem persists try a smaller date range or fewer projects."
-        elif isinstance(
-            error,
-            (snuba.UnqualifiedQueryError, snuba.QueryExecutionError, snuba.SchemaValidationError),
-        ):
-            message = "Internal error. Your query failed to run."
-        raise ExportError(message)
+# Adapted into decorator from 'src/sentry/api/endpoints/organization_events.py'
+def handle_snuba_errors(logger):
+    def wrapper(func):
+        @wraps(func)
+        def wrapped(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except discover.InvalidSearchQuery as error:
+                metrics.incr(
+                    "dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0
+                )
+                logger.info("dataexport.error: %s", six.text_type(error))
+                capture_exception(error)
+                raise ExportError("Invalid query. Please fix the query and try again.")
+            except snuba.QueryOutsideRetentionError as error:
+                metrics.incr(
+                    "dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0
+                )
+                logger.info("dataexport.error: %s", six.text_type(error))
+                capture_exception(error)
+                raise ExportError("Invalid date range. Please try a more recent date range.")
+            except snuba.QueryIllegalTypeOfArgument as error:
+                metrics.incr(
+                    "dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0
+                )
+                logger.info("dataexport.error: %s", six.text_type(error))
```

Comment:
You could skip the logs as you're capturing exceptions, which will give you way better information.

Reply:
These logs are quite useful in tests where (correct me if I'm wrong) there is no way to see the exception in sentry.

Reply:
Yes that is true.